sensing/pointcloud_preprocessor/include/pointcloud_preprocessor/outlier_filter/ring_outlier_filter_nodelet.hpp

```diff
@@ -45,18 +54,27 @@ class RingOutlierFilterComponent : public pointcloud_preprocessor::Filter
   rcl_interfaces::msg::SetParametersResult paramCallback(const std::vector<rclcpp::Parameter> & p);
 
   bool isCluster(
-    sensor_msgs::msg::PointCloud2::SharedPtr & input_ptr,
-    const std::vector<std::size_t> & tmp_indices)
+    const PointCloud2ConstPtr & input, std::pair<int, int> data_idx_both_ends, int walk_size,
+    const TransformInfo & transform_info)
   {
-    PointXYZI * front_pt = reinterpret_cast<PointXYZI *>(&input_ptr->data[tmp_indices.front()]);
-    PointXYZI * back_pt = reinterpret_cast<PointXYZI *>(&input_ptr->data[tmp_indices.back()]);
-
-    const auto x_diff = front_pt->x - back_pt->x;
-    const auto y_diff = front_pt->y - back_pt->y;
-    const auto z_diff = front_pt->z - back_pt->z;
-    return static_cast<int>(tmp_indices.size()) > num_points_threshold_ ||
-           (x_diff * x_diff) + (y_diff * y_diff) + (z_diff * z_diff) >=
-             object_length_threshold_ * object_length_threshold_;
+    if (walk_size > num_points_threshold_) return true;
+
+    auto first_point = reinterpret_cast<const PointXYZI *>(&input->data[data_idx_both_ends.first]);
+    auto last_point = reinterpret_cast<const PointXYZI *>(&input->data[data_idx_both_ends.second]);
+
+    Eigen::Vector4f p1(first_point->x, first_point->y, first_point->z, 1);
+    Eigen::Vector4f p2(last_point->x, last_point->y, last_point->z, 1);
+
+    if (transform_info.need_transform) {
+      p1 = transform_info.eigen_transform * p1;
+      p2 = transform_info.eigen_transform * p2;
+    }
```

Comment:
I don't think the transformation is necessary here:
Distance between 2 points is unchanged by translation/rotation:
`|T*p2 - T*p1| = |p2 - p1|`

Reply:
I made an ordinary mistake 💦  Thanks!